client/src/context/useBoardContext.tsx

```diff
@@ -1,25 +1,87 @@
-import { useState, useContext, createContext, FunctionComponent, useCallback } from 'react';
-import { Column } from '../interface/Column';
-import { columns } from '../mocks/mockBoardData';
+import { useState, useContext, createContext, FunctionComponent, useCallback, useEffect } from 'react';
+import { Column } from '../interface/Board';
+// import { columns } from '../mocks/mockBoardData';
+import getAllBoards from '../helpers/APICalls/getAllBoards';
+import getBoard from '../helpers/APICalls/getBoard';
+import createBoard from '../helpers/APICalls/createBoard';
+import { AllBoardResponse } from '../interface/BoardApiData';
 
 interface IBoardContext {
-  board: Column[];
+  boardList: AllBoardResponse[] | undefined;
+  boardColumns: Column[] | undefined;
+  boardName: string;
   updateBoard: (board: Column[]) => void;
+  switchBoardInView: (boardId: string) => void;
+  createNewBoard: (name: string) => void;
 }
 
 export const BoardContext = createContext<IBoardContext>({
-  board: [],
+  boardList: [],
+  boardColumns: [],
+  boardName: '',
   updateBoard: () => null,
+  switchBoardInView: () => null,
+  createNewBoard: () => null,
 });
 
 export const BoardProvider: FunctionComponent = ({ children }): JSX.Element => {
-  const [board, setBoard] = useState(columns);
+  const [boardColumns, setBoardColumns] = useState<Column[]>();
+  const [boardList, setBoardList] = useState<AllBoardResponse[]>([]);
+  const [boardName, setBoardName] = useState<string>('');
 
-  const updateBoard = useCallback((newBoard: Column[]) => {
-    setBoard(newBoard);
+  useEffect(() => {
+    const setInitialBoardList = async () => {
+      const allboardResponse = await getAllBoards();
+      if (allboardResponse.success) {
+        const { boardsList } = allboardResponse.success;
+        setBoardList(boardsList);
+        const lastViewedBoard = boardsList.find((e: AllBoardResponse) => e.lastViewed === true);
+        if (lastViewedBoard) {
+          const lastViewedBoardDetails = await getBoard(lastViewedBoard._id);
+          if (lastViewedBoardDetails.success) {
+            const { board } = lastViewedBoardDetails.success;
+            setBoardColumns(board.columns);
+            setBoardName(board.name);
+          }
+        }
+      }
+    };
+    setInitialBoardList();
   }, []);
 
-  return <BoardContext.Provider value={{ board, updateBoard }}>{children}</BoardContext.Provider>;
+  // IGOR IS WORKING ON THIS
+  const updateBoard = useCallback(() => {
+    // setBoard(newBoard);
+  }, []);
+
+  const switchBoardInView = useCallback(async (boardId) => {
```

Comment:
Could you elaborate on your reason for `useCallback` here?

Reply:
Using `useCallback` loosely actually causes performance issues rather than solves it. I would advise to use it only when necessary and clear performance boosts to use it.

Reply:
You can read more on it from this article https://kentcdodds.com/blog/usememo-and-usecallback